### PR TITLE
chore: reorganize benchmarks

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

The benchmark files scattered in `babel-generator`, `babel-helper-validator-identifier` and `babel-parser` are now moved to a top level `/benchmark` folder. They now share the same `/util.mjs` benchmark setup and we don't have to hardcode package index.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13606"></a>

### CODE REVIEW

File: benchmark/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "name": "@babel/benchmark",
+  "private": true,
+  "type": "module",
+  "devDependencies": {
+    "@babel-baseline/generator": "npm:@babel/generator@7.14.5",
+    "@babel-baseline/helper-validator-identifier": "npm:@babel/helper-validator-identifier@7.10.4",
+    "@babel-baseline/parser": "npm:@babel/parser@7.14.5",
+    "@babel/generator": "workspace:*",
+    "@babel/helper-validator-identifier": "workspace:*",
+    "@babel/parser": "workspace:*",
```
Comment (JLHwung) [inline benchmark/package.json:11]: Now we can compare performances between any versions of babel packages by revising this `package.json`.

**Review by fedeci — APPROVED** [review]
Do we have any report tool that runs benchmarks along with tests on the CI?

**Review by nicolo-ribaudo — APPROVED** [review]

**Comment by JLHwung** [comment]
> Do we have any report tool that runs benchmarks along with tests on the CI?

Not yet. After we centralize the benchmark files, we can come up with a benchmark bot that plots benchmark alongside versions so we can catch performance regressions from new releases.

File: benchmark/package.json (hunk 1 above)
Comment (KFlash) [reply benchmark/package.json:11]: what's the perf diff between ESBuild and Babel? Equal in performance?

**Pull request merged**